tests/test_api.py

```diff
@@ -0,0 +1,37 @@
+import pytest
+import pyux
+import keras
+import json
+import os
+
+import keras.backend.tensorflow_backend
+import keras.backend.theano_backend
+import keras.backend.cntk_backend
+import keras.backend.numpy_backend
+import keras.utils.test_utils
+
+
+def test_api():
+    api_file = os.path.join(os.getcwd(), 'api.json')
+
+    with open(api_file, 'r') as f:
+        previous_api = json.load(f)
+
+    current_api = pyux.sign(keras)
+
+    diff = pyux.diff(current_api, previous_api)
+
+    exceptions = [
+        pyux.ADDED_ARG_WITH_DEFAULT_IN_METHOD,
+        pyux.ADDED_DEFAULT_IN_METHOD
+    ]
+
+    diff = list(filter(lambda c: c[0] not in exceptions, diff))
+
+    if diff:
+        str_diff = '\n'.join([str(x) for x in diff])
+        raise Exception("API change detected ! \n " + str_diff)
```

Comment:
Could we get an item-by-item print of all differences here? Comparing e.g. the old signature to the new signature for each symbol that has changed.

Reply:
`diff` is a list of tuples, each tuple of form `(Type of change, Item which changed, before value, after value)`. Would you like to see some sample diffs?

Reply:
Sure, seeing some sample diffs would be great!

Btw is there a more meaningful exception type we could raise instead of `Exception`?

Reply:
Resolved - we use `pyux.APIChangedException` instead.